documentation/src/main/asciidoc/user_guide/eviction.adoc

```diff
@@ -87,19 +109,38 @@ Programmatically, the same would be defined using:
 [source,java]
 ----
 Configuration c = new ConfigurationBuilder().eviction().strategy(EvictionStrategy.LRU)
-               .maxEntries(2000).expiration().wakeUpInterval(5000l).lifespan(1000l).maxIdle(500l)
+               .size(2000).expiration().wakeUpInterval(5000l).lifespan(1000l).maxIdle(500l)
                .build();
 
 ----
 
+==== Memory Based Eviction Configuration
+Memory based eviction may require some additional configuration options if you
+are using your own custom types as Infinispan is normally used.  In this case
+Infinispan cannot estimate the memory usage of your classes and as such you are
+required to use `storeAsBinary` when memory based eviction is used.
```

Comment:
1. Updated all of the paragraphs to have new line
2. Sure
3. It throws an exception that isn't the clearest, but gets the job done :P
4. Added.